### DISCUSSION TITLE
OnGUI window call

### DISCUSSION THREAD

**Discussion by Patrick-van-Halm** [vfsfitvnm/frida-il2cpp-bridge#445] [Q&A]
I am trying to create a window using the IMGUI methods. Currently I have the following: 
```js
public static Window(id: number, rect: Rect, func: (id: number) => void, title: string, style: Il2Cpp.Object): void {
        const delegate = UnityEngineIMGUIModuleImage.class("UnityEngine.GUI").nested("WindowFunction");
        return GUI.Definition.Invoke("Window", id, rect.unbox(), Il2Cpp.delegate(delegate, func), Il2Cpp.string(title), style);
}
```

But this results in the following error:
```
Error: access violation accessing 0xffffffffffffffff
    at invokeRaw (:6)
    at invoke (:6)
    at r (FridaExt.ts:50)
    at Window (GUI.ts:29)
    at <anonymous> (index.ts:18)
    at call (native)
    at <anonymous> (:6)
```

How could I solve this?
I am also curious how I could force OnGUI to run when there isn't a class with an OnGUI implementation. Currently I add a monobehaviour in the scene that contains an OnGUI method and change the implementation. Is there a better way?

**Comment by VAROME** [comment]
https://github.com/maarsalien/frida-android-mod-menu

**Reply by Patrick-van-Halm** [reply]
Thanks but how does this help me? I am using it not on Android. This is for a windows Unity3d game...

So I am looking for something that runs only within Unity.

**Reply by HenryNotRich** [reply]
create similar to this 😹
